src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -152,9 +153,11 @@ export default class DefaultDeviceController
   private inputDeviceCount: number = 0;
   private lastNoVideoInputDeviceCount: number;
 
-  private browserBehavior: DefaultBrowserBehavior = new DefaultBrowserBehavior();
-
-  constructor(private logger: Logger, options?: { enableWebAudio?: boolean }) {
+  constructor(
+    private logger: Logger,
+    options?: { enableWebAudio?: boolean },
+    private browserBehavior: ExtendedBrowserBehavior = new DefaultBrowserBehavior()
```

Comment:
This can be optional as well or since we are providing `DefaultBrowserBehavior` object it is not a breaking change, correct?

Could you please explain in what cases we want to override and have this option in constructor? In the description you have said "This fixes that important omitted extension point.", but could not understand it.

Reply:
I see you have a unit test with this case, but trying to understand from builders perspective is this required?

Reply:
This is an optional argument — that's implied by the `=`. And yes, it's not a breaking change because we are defaulting to `DBB`.

I have _almost_ made this change a bunch of times, because I kept needing to supply different behavior — e.g., this would eliminate the need for much of our DOM mocking in tests!

A builder might provide the argument here if they want a different implementation of any of the methods in `BB`. For example, they might know that their transform devices don't use Audio Worklet, and want to return `false` from the new method I added because they don't need this workaround.

Or they might want to work around a bug in the SDK's feature detection (e.g., to insist that screen share is supported) or force a specific video codec.

The alternative to this is that they have to implement their own `DeviceController`, or break encapsulation and poke at the internals of `DDC`.

Reply:
Yup, I agree with the flexibility this introduces. But, if builders provide their `DBB` implementation to the `DDC` then the same browser behavior should be passed to other classes like the `DAV` cause, now if any method implementation contradicts then the behavior would be different and a flow may run into error, right? We have used different instances of `DBB` for e.g. [1](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/audiovideocontroller/DefaultAudioVideoController.ts#L122), [2](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/audiovideocontroller/DefaultAudioVideoController.ts#L235) which is why I am little cautious here.

Reply:
If we are currently relying on asking the same question in two or three places and getting the same answer, we should probably not do that.

Yes, we should allow passing the customer's provided behavior implementation around. No, I don't intend to do that in this PR because it'll keep getting bigger :)

Reply:
So would it be better to do the constructor change in `DDC` when we actually update the rest of the classes which need `DBB`? I agree that it would be a bigger PR and can be taken up later as well as we really need to use `DBB` as a singleton to avoid this issue.

Reply:
No, because a customer who wants to opt out of this workaround needs a way to do so.

Reply:
(Also: no singletons!)

Reply:
Yeah it should be same object no singleton. Thanks for clarifying the use here.